File: 2022/puzzle_1.rb
calories_per_elf = File.open("puzzle_input_1.txt") do |f|
    f.readlines(chomp: true)
        .slice_when { |i, line| line.empty? }
        .map { |groups| groups.map(&:to_i).sum }
end

most_calories = calories_per_elf.max
top_three = calories_per_elf.sort.reverse.take(3).sum

File: 2022/puzzle_1.py
import itertools

with open("puzzle_input_1.txt") as f:
    lines = f.read().splitlines()

    groups = [list(g) for k, g in itertools.groupby(lines, lambda l: len(l))]
    filtered = filter(lambda grp: len(grp[0]), groups)
    calories_per_elf = list(map(lambda grp: sum(map(int, grp)), filtered))

    most_calories = max(calories_per_elf)
    top_three_summed = sum(sorted(calories_per_elf, reverse=True)[0:3])